client/jetpack-cloud/sections/partner-portal/issue-license-form/index.tsx

```diff
@@ -21,6 +22,12 @@ function selectProductOptions( families: APIProductFamily[] ): APIProductFamilyP
 	return families.flatMap( ( family ) => family.products );
 }
 
+function alphabeticallySortedProductOptions(
+	families: APIProductFamily[]
+): APIProductFamilyProduct[] {
+	return sortBy( selectProductOptions( families ), ( product ) => product.name );
+}
```

Comment:
Just added a composite selector to do the alphabetical sorting. Opted for doing this to be easier to decouple this logic whenever we implement a broader redesign for this page.